app/react/kubernetes/queries/useEvents.ts

```diff
@@ -27,22 +32,15 @@ async function getNamespaceEvents(
   }
 }
 
-export function useNamespaceEventsQuery(
+export function useEvents(
   environmentId: EnvironmentId,
-  namespace: string,
+  namespace?: string,
   options?: { autoRefreshRate?: number },
-  labelSelector?: string
+  params?: Params
 ) {
   return useQuery(
-    [
-      'environments',
-      environmentId,
-      'kubernetes',
-      'events',
-      namespace,
-      labelSelector,
-    ],
-    () => getNamespaceEvents(environmentId, namespace, labelSelector),
+    ['environments', environmentId, 'kubernetes', namespace, 'events', params],
```

Comment:
why don't you have a query keys factory, and a URL builder function?

Reply:
url builder seems like over engineering in this case. I do see the point in query keys, so have added that

Reply:
it's not over-engineering, it's making the code consistent. that's how we do it in most of the code base